src/Storages/StorageSnapshot.cpp

```diff
@@ -112,33 +112,47 @@ NameAndTypePair StorageSnapshot::getColumn(const GetColumnsOptions & options, co
     return *column;
 }
 
-Block StorageSnapshot::getSampleBlockForColumns(const Names & column_names) const
+Block StorageSnapshot::getSampleBlockForColumns(const Names & column_names,const NameToNameMap & parameter_values) const
 {
     Block res;
+
     const auto & columns = getMetadataForQuery()->getColumns();
-    for (const auto & name : column_names)
+    for (const auto & column_name : column_names)
     {
-        auto column = columns.tryGetColumnOrSubcolumn(GetColumnsOptions::All, name);
-        auto object_column = object_columns.tryGetColumnOrSubcolumn(GetColumnsOptions::All, name);
+        /// substituted_column_name is used for parameterized view (which are created using query parameters
+        /// and SELECT is used with substitution of these query parameters )
+        std::string substituted_column_name = column_name;
+        std::string::size_type pos = 0u;
+        for (const auto & parameter : parameter_values)
```

Comment:
Not necessary but this could be move into a separate a separate function/method (for example called `replaceQueryParameterValue`), so the code will be like: 
```
std::string substituted_column_name
if (parameter_values.empty())
    substituted_column_name = column_name;
else
    substituted_column_name = replaceQueryParameterValue(column_name, parameter_values);
```
 just for better readability

Reply:
good point, added 2 static functions for column name manipulation in StorageView